src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -345,6 +345,110 @@ void QgsPostgresProvider::setTransaction( QgsTransaction *transaction )
   mTransaction = static_cast<QgsPostgresTransaction *>( transaction );
 }
 
+struct Ewkt
+{
+  int srid = -1;
+  QString wkt;
+};
+
+QgsReferencedGeometry QgsPostgresProvider::fromEwkt( const QString &ewkt, QgsPostgresConn *conn )
+{
+  thread_local const QRegularExpression regularExpressionSRID( "^SRID=(\\d+);" );
+
+  QRegularExpressionMatch regularExpressionMatch = regularExpressionSRID.match( ewkt );
+  if ( !regularExpressionMatch.hasMatch() )
+    return QgsReferencedGeometry();
+
+  Ewkt ewktInfo;
+  ewktInfo.wkt = ewkt.mid( regularExpressionMatch.captured( 0 ).size() );
+  ewktInfo.srid = regularExpressionMatch.captured( 1 ).toInt();
+
+
+  QgsGeometry geom = QgsGeometry::fromWkt( ewktInfo.wkt );
+  return QgsReferencedGeometry( geom, sridToCrs( ewktInfo.srid, conn ) );
+}
+
+QString QgsPostgresProvider::toEwkt( const QgsReferencedGeometry &geom, QgsPostgresConn *conn )
+{
+  return QStringLiteral( "SRID=%1;%2" ).arg( QString::number( crsToSrid( geom.crs(), conn ) ), geom.asWkt() );
+}
+
+QString QgsPostgresProvider::geomAttrToString( const QVariant &attr, QgsPostgresConn *conn )
+{
+  if ( attr.type() == QVariant::String )
+    return attr.toString();
+  else
+    return toEwkt( attr.value<QgsReferencedGeometry>(), conn );
+}
+
+static QMutex sMutex;
+static QMap<int, QgsCoordinateReferenceSystem> sCrsCache;
```

Comment:
Shouldn't this be per-connection? It's possible that a user is using two different databases with different spatial ref table values

Reply:
Yes, but I see this as a separate bugfix (this just keeps the status quo)